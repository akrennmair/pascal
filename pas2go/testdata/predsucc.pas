program test;

type ttt = (foo, bar, baz);

var x : ttt;
	i : integer;

begin
	x := foo;
	writeln('x = ', x);
	x := succ(x);
	writeln('x = ', x);
	x := succ(x);
	writeln('x = ', x);

	x := pred(x);
	writeln('x = ', x);
	x := pred(x);
	writeln('x = ', x);

	i := 2;
	i := succ(i);
	writeln('i = ', i);
	i := pred(i);
	writeln('i = ', i)
end.
