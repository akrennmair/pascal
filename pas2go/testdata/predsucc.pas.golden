package main

import (
	"github.com/akrennmair/pascal/pas2go/system"
)

var _ = system.Write

// program test
func main() {
	type (
		ttt int
	)
	const (
		foo ttt = 0
		bar ttt = 1
		baz ttt = 2
	)

	var (
		x ttt
		i int
	)
	_ = x
	_ = i

	x = foo
	system.Writeln("x = ", x)
	x = (x + 1)
	system.Writeln("x = ", x)
	x = (x + 1)
	system.Writeln("x = ", x)
	x = (x - 1)
	system.Writeln("x = ", x)
	x = (x - 1)
	system.Writeln("x = ", x)
	i = 2
	i = (i + 1)
	system.Writeln("i = ", i)
	i = (i - 1)
	system.Writeln("i = ", i)
}
